File: Lesson 3 Time Complexity/TapeEquilibrium.py
def solution(A):
    sum_after = sum(A[1:])
    sum_before = A[0]
    min_sum = abs(sum_after - sum_before)
    for i in range(1,len(A)-1):
        sum_after -= A[i]
        sum_before += A[i]

        if min_sum > abs(sum_after - sum_before):
            min_sum = abs(sum_after - sum_before)
    return min_sum
	
def solution2(A):
    sum_before = A[0]
    sum_after = sum(A[1:])
    min_sum = abs(sum_before - sum_after)
    for i in A[1:-1]:
        sum_after -= i 
        sum_before += i

        
        if min_sum > abs(sum_before - sum_after):
            min_sum = abs(sum_before - sum_after)
    return min_sum

File: Lesson 3 Time Complexity/TapeEquilibrium.c
int solution(int A[], int N) {
    
    int sum_after = 0;
    int sum_before = A[0];
    for (int i=1;i<N;i++)
    {
        sum_after += A[i];
    }

    int min_sum = abs(sum_after - sum_before);

    for(int i=1;i<N-1;i++)
    {
        sum_after -= A[i];
        sum_before += A[i];

        if(min_sum > abs(sum_after - sum_before))
            min_sum = abs(sum_after - sum_before);
    }
    return min_sum;
}